src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -316,9 +319,12 @@ protected void install() {
                     return;
                 }
             }
-            stopSyncingShadows(false);
-            syncHandler.setSyncStrategy(strategy);
-            startSyncingShadows(StartSyncInfo.builder().build());
+            if (!currentStrategy[0].equals(strategy)) {
```

Comment:
Does this cover the other case where we are going from the Periodic (non-default) to RealTime (default) strategy? It looks like here that currentStrategy will always be (DEFAULT_STRATEGY) and if we convert back the code block doesn't get executed.

Reply:
whenever the strategy is updated we store a reference to it and then compare it to the new one

In this case, we start at DEFAULT if there is no config and store that. Then we update to periodic, and that replaces the value. Then we switch back to DEFAULT and that again replaces the stored value.